source/optouts/OptOutsService.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Web;
+using com.esendex.sdk.core;
+using com.esendex.sdk.exceptions;
+using com.esendex.sdk.optouts.models;
+using com.esendex.sdk.utilities;
+
+namespace com.esendex.sdk.optouts
+{
+    public class OptOutsService
+    {
+        private const string OPTOUTS_BASE_URL = "https://api.esendex.com";
+
+        private readonly Uri _baseUrl;
+        private readonly EsendexCredentials _credentials;
+
+        internal OptOutsService(string url, EsendexCredentials esendexCredentials)
+        {
+            _baseUrl = new Uri(url);
+            _credentials = esendexCredentials;
+        }
+
+        public OptOutsService(EsendexCredentials credentials) : this(OPTOUTS_BASE_URL, credentials)
+        {
+        }
+
+        public OptOutsService(string username, string password) : this(OPTOUTS_BASE_URL, new EsendexCredentials(username, password))
+        {
+        }
+
+        public OptOut GetById(Guid optOutId)
+        {
+            var requestUrl = new Uri(_baseUrl, string.Format("v1.0/optouts/{0}", optOutId));
+            var request = Request.Create("GET", requestUrl)
+                                 .WithHeader("Authorization", "Basic " + _credentials.EncodedValue())
+                                 .WithAcceptHeader(Constants.XML_MEDIA_TYPE)
+                                 .If(_credentials.UseProxy, r => r.WithProxy(_credentials.WebProxy));
+
+            try
+            {
+                var response = request.GetResponse();
+                using (var reader = new StreamReader(response.GetResponseStream(), Encoding.UTF8))
+                {
+                    string responseText = reader.ReadToEnd();
+                    return new XmlSerialiser().Deserialise<OptOut>(responseText);
+                }
+            }
+            catch (WebException ex)
+            {
+                var response = (HttpWebResponse)ex.Response;
+                if (response.StatusCode != HttpStatusCode.BadRequest)
+                    throw;
+
+                throw new BadRequestException(ex, null);

```

Comment:
I spoke to Dave about this and he said not to bother with structured errors, though maybe he thought it would be more work than it was. I just followed through with that but will be happy to add it if necessary.